modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java

```diff
@@ -224,7 +224,11 @@ public RustClientCodegen() {
                         Map<String, Object> mas = new HashMap<>();
                         mas.put("modelName", camelize(mappedModel.getModelName()));
                         mas.put("mappingName", mappedModel.getMappingName());
-                        List<CodegenProperty> vars = model.getVars();
+                        
+                        // TODO: deleting the variable from the array was
+                        // problematic; I don't know what this is supposed to do
+                        // so I'm just cloning it for the moment
+                        List<CodegenProperty> vars = new ArrayList<>(model.getVars());
```

Comment:
@ahl when you've time, can you please elaborate on the problem ?

Reply:
Yeah, sorry to leave that turd. This code section seems to be handling discriminators and relocates a particular variable into the discriminators variables. It previously did so destructively which caused the original variable to be absent.

This is definitely ugly and poorly structured, but I don't have a good sense of how it should be structured or how to properly and exhaustively test some of the subtle semantics here.

Reply:
No worries bro. Do you have a spec that can reproduce the issue that you encountered with this line/section of code?

Reply:
I was running the influxdb openapi spec through it: https://docs.influxdata.com/influxdb/v2.0/api/

Reply:
OK. Will take a look when I can find the time later.